homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,210 @@
+"""Config flow for ezviz."""
+import logging
+
+from pyezviz import EzvizClient
+from pyezviz.client import PyEzvizError
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_PASSWORD, CONF_REGION, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import (  # pylint: disable=unused-import
+    ATTR_SERIAL,
+    CONF_FFMPEG_ARGUMENTS,
+    DEFAULT_CAMERA_USERNAME,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_REGION,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ezviz."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    async def _validate_and_create_auth(self, data):
+        """Validate the user input allows us to connect.
+
+        Data has the keys from data_schema with values provided by the user.
+        """
+        await self.async_set_unique_id(data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        # constructor does login call
+        client = EzvizClient(
+            data[CONF_USERNAME],
+            data[CONF_PASSWORD],
+            data.get(CONF_REGION, DEFAULT_REGION),
+            data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        )
+
+        # Validate data by sending a login request to ezviz
+
+        try:
+            await self.hass.async_add_executor_job(client.login)
+
+        except PyEzvizError as err:
+            raise InvalidAuth from err
+
+        auth_data = {
+            "username": data[CONF_USERNAME],
+            "password": data[CONF_PASSWORD],
+            "region": data.get(CONF_REGION, DEFAULT_REGION),
+        }
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
+
+    async def _create_camera_rstp(self, data):
+        """Create RSTP auth entry per camera in config."""
+
+        await self.async_set_unique_id(data[ATTR_SERIAL])
+        self._abort_if_unique_id_configured()
+
+        _LOGGER.debug("Create camera with: %s", data)
+
+        return self.async_create_entry(title=data[ATTR_SERIAL], data=data)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return EzvizOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
```

Comment:
Are we sure that the existing entry(ies) is the main entry? What if the user deletes the main entry and runs the config flow again?

Reply:
Added "type" value under data, check if account type is present before evaluating which form to present the user with.

Reply:
Remove the check here. Just iterate.